device-posture/device-posture.https.html

```diff
@@ -0,0 +1,86 @@
+<script>
+'use strict';
+
+const originalPosture = navigator.devicePosture.type;
+async function cleanup() {
+  await test_driver.set_device_posture(originalPosture);
+}
+
+promise_test(async (t) => {
+  t.add_cleanup(cleanup);
+  const watcher = new EventWatcher(t, navigator.devicePosture, ['change']);
+  const postures = ['folded', 'continuous', 'folded'];
+  for (const posture of postures) {
+    test_driver.set_device_posture(posture);
+    await watcher.wait_for('change').then(() => {
+      assert_equals(posture, navigator.devicePosture.type);
+    });
+  }
+}, 'Device posture test.');
+
+
+
+
+
+// REFERENCE https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/http/tests/inspector-protocol/device-posture/device-posture-change-event.js
+promise_test(async (t) => {
+  t.add_cleanup(cleanup);
+  assert_equals(navigator.devicePosture.type, 'continuous');
+
+  const promise = new Promise(resolve => {
+    navigator.devicePosture.onchange = function() {
+        resolve(navigator.devicePosture.type);
+      }
+  });
+  test_driver.set_device_posture('folded');
+  assert_equals(await promise, 'folded')
+}, 'Device posture test: device-posture-change-event.js.');
+
+// REFERENCE https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/http/tests/inspector-protocol/device-posture/device-posture-default.js
+promise_test(async (t) => {
+  t.add_cleanup(cleanup);
+  assert_equals(navigator.devicePosture.type, 'continuous');
+  await test_driver.set_device_posture('folded');
+  assert_equals(navigator.devicePosture.type, 'folded')
+}, 'Device posture test: device-posture-default.js.');
+
+// REFERENCE https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/http/tests/inspector-protocol/device-posture/device-posture-event-listener.js
+promise_test(async (t) => {
+  t.add_cleanup(cleanup);
+  assert_equals(navigator.devicePosture.type, 'continuous');
+
+  const promise = new Promise(resolve => {
+    navigator.devicePosture.addEventListener(
+      'change',
+      () => { resolve(navigator.devicePosture.type); },
+      { once: true }
+    );
+  });
+  test_driver.set_device_posture('folded');
+  assert_equals(await promise, 'folded')
+}, 'Device posture test: device-posture-event-listener.js.');
+
+// REFERENCE https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/http/tests/inspector-protocol/device-posture/device-posture-media-queries.js
+promise_test(async (t) => {
+  t.add_cleanup(cleanup);
+  assert_equals(navigator.devicePosture.type, 'continuous');
+  assert_true(matchMedia('(device-posture: continuous)').matches);
+
+  const foldedMQL = window.matchMedia('(device-posture: folded)');
+  const promise = new Promise(resolve => {
+    foldedMQL.addEventListener(
+      'change',
+      () => { resolve(foldedMQL.matches); },
+      { once: true }
+    );
+  });
+  test_driver.set_device_posture('folded');
```

Comment:
You're not handling the promise here either.

Reply:
What does "not handling the promise" in here means? 
Isn't "await promise" in line 83 waiting the event to happen. And also if "resolve(foldedMQL.matches);" line is executed ==> the promise is handled?

Reply:
"The promise" isn't the same as "`promise`". You are handling `promise` in line 84 by `await`ing on it, but I'm referring to _the promise_ returned by `test_driver.set_device_posture()` in this line. In the current version of the code, nothing guarantees that this promise has resolved by the time the test finishes and that it has run successfully.

Reply:
OK , the promise of test_driver.set_device_posture call. 

Learning the asynchronous nature of this language/Chromium env is ongoing. I was thinking that if you wait test_driver.set_device_posture to complete the event "assert_true(await promise)" is waiting would happen before we are in that line (so it would "miss" the event). Also "assert_true(await promise)" would guarantee that test cannot finish without device-posture to be  folded.

Reply:
Fixed in https://github.com/JuhaVainio/WPT/pull/4/commits/3831e1ba726cd5fad98c5de482ade8d24222fe12. The first test will be fixed in another commit